app/views/components/_email-signup-section.html.erb

```diff
@@ -0,0 +1,45 @@
+<%
+  details ||= {}
+  link_text ||= false
+  link_href ||= false
+  intro ||= false
+  full_width ||= false
+  track_action ||= false
+  track_label ||= false
+  transparent_background ||= false
+  css_classes ||= false
```

Comment:
We don't allow arbitrary classes to be passed to components because it destroys isolation (an application could style a component, which would make maintaining that component very difficult). Obviously this component is specific to this application, but I think this still applies. What's the use case here?

We do allow classes to be passed that are prefixed with `js-` for JS hooks, see the shared_helper in the components gem for how that's handled.

Reply:
Incidentally you can use the shared helper from the gem, in the same way that it works for components in the gem...

`shared_helper = GovukPublishingComponents::Presenters::SharedHelper.new(local_assigns)`

You can then use all of the functionality of the helper, including the `classes` functionality for accepting only `js-` prefixed classes.

Reply:
@andysellick yep makes sense. Thanks for suggesting the shared helper, it's good to know that functionality is available! However I think we do not need it after all – after my most recent updates there is no need to pass through custom classes anymore.

Reply:
Might you also want it for the `margin_bottom` functionality it provides?